packages/leancode_analytics/lib/core/route_observer.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:leancode_analytics/leancode_analytics.dart';
+
+/// Interface that should be impletemented by default app [Route]
+abstract interface class LeanAnalyticsRoute {
+  /// Required identifier to distinguish page Routes
+  String get id;
+
+  /// Optional [analyticsParams] to pass with [ScreenViewAnalyticsEvent]
+  Map<String, dynamic>? get analyticsParams;
+}
+
+/// [NavigatorObserver] that register event when [LeanAnalyticsRoute] shows
+/// on the screen
+class LeanAnalyticsNavigationObserver extends NavigatorObserver {
+  /// Creates a [LeanAnalyticsNavigationObserver] with [analytics]
+  LeanAnalyticsNavigationObserver(this.analytics);
+
+  /// Instance of [LeanAnalytics] to register events on [Route] changes
+  final LeanAnalytics analytics;
+
+  @override
+  void didPush(Route<dynamic> route, Route<dynamic>? previousRoute) {
+    if (route case LeanAnalyticsRoute _) {
```

Comment:
`case final` to avoid casting (you seem to already be doing it in other callbacks)